zerver/tests/test_subs.py

```diff
@@ -1042,6 +1042,46 @@ def test_change_history_access_for_private_streams(self) -> None:
         ).decode()
         self.assertEqual(realm_audit_log.extra_data, expected_extra_data)
 
+    def test_add_and_remove_stream_as_default(self) -> None:
+        user_profile = self.example_user("hamlet")
+        self.login_user(user_profile)
+        realm = user_profile.realm
+        default_stream = self.make_stream("default_stream", realm=realm)
+
+        do_change_user_role(user_profile, UserProfile.ROLE_REALM_ADMINISTRATOR, acting_user=None)
+        params = {
+            "is_default_stream": orjson.dumps(True).decode(),
+        }
+        default_stream_id = self.subscribe(user_profile, "default_stream").id
+        result = self.client_patch(f"/json/streams/{default_stream_id}", params)
+        self.assert_json_success(result)
+        self.assertTrue(default_stream in get_default_streams_for_realm(realm.id))
+
+        params = {
+            "is_default_stream": orjson.dumps(False).decode(),
+        }
+        result = self.client_patch(f"/json/streams/{default_stream_id}", params)
+        self.assert_json_success(result)
+        self.assertFalse(default_stream in get_default_streams_for_realm(realm.id))
+
+        private_stream = self.make_stream("private_stream", realm=realm, invite_only=True)
+        private_stream_id = self.subscribe(user_profile, "private_stream").id
+        params = {
+            "is_default_stream": orjson.dumps(True).decode(),
+        }
+        result = self.client_patch(f"/json/streams/{private_stream_id}", params)
+        self.assert_json_error(result, "Private streams cannot be made default.")
+        self.assertFalse(private_stream in get_default_streams_for_realm(realm.id))
+
+        default_stream_2 = self.make_stream("default_stream_2")
+        do_change_user_role(user_profile, UserProfile.ROLE_MEMBER, acting_user=None)
+        params = {
+            "is_default_stream": orjson.dumps(True).decode(),
+        }
+        result = self.client_patch(f"/json/streams/{default_stream_2.id}", params)
+        self.assertFalse(default_stream_2 in get_default_streams_for_realm(realm.id))
```

Comment:
You might also need to add tests for checking case of passing `is_default_stream: True` and `is_private: True`.